main.tf

```
@@ -473,3 +475,25 @@ resource "aws_secretsmanager_secret_rotation" "this" {
     schedule_expression      = var.master_user_password_rotation_schedule_expression
   }
 }
+
+################################################################################
+# RDS Shard Group
+################################################################################
+
+resource "aws_rds_shard_group" "this" {
+  count = local.create && var.create_shard_group ? 1 : 0
+
+  compute_redundancy        = var.compute_redundancy
+  db_cluster_identifier     = aws_rds_cluster.this[0].id
+  db_shard_group_identifier = var.db_shard_group_identifier
+  max_acu                   = var.max_acu
+  min_acu                   = var.min_acu
+  publicly_accessible       = var.publicly_accessible
+  tags                      = var.tags
```

```suggestion
  tags                      = merge(var.tags, var.shard_group_tags)
```
or something like it...